nixos/modules/services/x11/xidlehook.nix

```diff
@@ -0,0 +1,225 @@
+            canceller = mkOption {
+              description = ''
+                The shell command invoked when the user becomes active after
+                the timer has gone off, but before the next timer (if any).
+                Optional.
+              '';
+              type = types.str;
+              default = "";
+            };
+          };
+        }));
+      };
+      extraOptions = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "--socket" "\${XDG_RUNTIME_DIR}/xidlehook.socket" "--not-when-fullscreen" "--not-when-audio" ];
+        description = ''
+          Additional command-line arguments to pass to
+          <command>xidlehook</command>.
+        '';
+      };
+
+
+      # Options for drop-in compatibility with xautolock
+
+      enableNotifier = mkEnableOption "xidlehook.notify" // {
+        description = ''
+          This publishes a notification before the autolock.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      time = mkOption {
+        default = 15;
+        type = types.int;
+
+        description = ''
+          Idle time (in minutes) to wait until xidlehook locks the computer.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      locker = mkOption {
+        default = "${pkgs.xlockmore}/bin/xlock"; # default according to `man xautolock`
+        example = "${pkgs.i3lock}/bin/i3lock -i /path/to/img";
+        type = types.str;
+
+        description = ''
+          The script to use when automatically locking the computer.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      notify = mkOption {
+        default = 10;
+        type = types.int;
+
+        description = ''
+          Time (in seconds) before the actual lock when the notification about the pending lock should be published.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      notifier = mkOption {
+        default = null;
+        example = "${pkgs.libnotify}/bin/notify-send \"Locking in ${cfg.notify} seconds\"";
+        type = types.nullOr types.str;
+
+        description = ''
+          Notification script to be used to warn about the pending autolock.
+
+          This option is for xautolock-configuration compatibility & has no effect if
+          services.xserver.xidlehook.timers is set directly.
+        '';
+      };
+
+      killer = mkOption {
+        default = null; # default according to `man xautolock` is none
+        example = "${pkgs.systemd}/bin/systemctl suspend";
```

Comment:
Needs `exampleText = "\${pkgs.systemd}/bin/systemctl suspend"`, same for other `default`'s that have `${}` in them

Reply:
What is `exampleText`?

I tried adding \ to the $s in these examples.  They still didn't render well in the manual:

-Example: ''''${pkgs.libnotify}/bin/notify-send "Locking in 10 seconds"''
+Example: ''''${pkgs.libnotify}/bin/notify-send "Locking in ''${toString cfg.notify} seconds"''

So I switched them all to literalExample.  Thanks!

Reply:
Ah sorry, I mixed it up with `defaultText`. Looking good now